packages/components/scripts/wrapper-generator/VueWrapperGenerator.ts

```diff
@@ -0,0 +1,144 @@
+  }
+
+  public generateProps(component: TagName, rawComponentInterface: string): string {
+    const propsName = this.generatePropsName(component);
+
+    return getComponentMeta(component).props
+      ? `  type ${propsName} = {${rawComponentInterface
+          .slice(1, -1)
+          .split(';\n')
+          .filter((x) => !x.match(/ {2}on[A-Z][a-z]+.+/))
+          .join(';\n')}  }`
+      : '';
+  }
+
+  public generateComponent(component: TagName, extendedProps: ExtendedProp[]): string {
+    const propsName = this.generatePropsName(component);
+    const eventNamesAndTypes = extendedProps
+      .filter(({ isEvent }) => isEvent)
+      .map(({ key, rawValueType }) => {
+        const [, type] = /<(\w+)>/.exec(rawValueType) ?? [];
+        return {
+          eventName: camelCase(key.replace('on', '')),
+          type,
+        };
+      });
+
+    const disableEslintForLine = ' // eslint-disable-line vue/require-valid-default-prop';
+
+    const defaultPropsWithValue = extendedProps
+      .map(({ key, defaultValue, isEvent }) => {
+        // TODO: better approach to identify object?
+        if (!(isEvent || defaultValue === undefined)) {
+          const transformedDefaultValue = defaultValue.startsWith('{') ? `() => (${defaultValue})` : defaultValue;
+          return `${key}: ${transformedDefaultValue},${
+            component === 'p-headline' && key === 'color' ? disableEslintForLine : ''
+          }`;
+        } else {
+          return undefined;
+        }
+      })
+      .filter((x) => x)
+      .join('\n     ');
+
+    const componentMeta = getComponentMeta(component);
+    const hasEvent = componentMeta.hasEvent;
+    const hasProps = componentMeta.props;
+    const hasSlot = componentMeta.hasSlot;
+    const hasDefaultProps = defaultPropsWithValue.length;
+
+    const defineProps = `defineProps<${propsName}>()`;
+    const props = `const props = ${
+      hasDefaultProps
+        ? `withDefaults(${defineProps}, {
+     ${defaultPropsWithValue}
+    })`
+        : defineProps
+    };`;
+
+    const defineEmits = `
+  const emit = defineEmits<{ ${eventNamesAndTypes
+    .map(({ eventName, type }) => `(e: '${eventName}', value: ${type}): void;`)
+    .join(' ')} }>();`;
+
+    const addEventListener = eventNamesAndTypes
+      .map(({ eventName }, index) => {
+        const { eventName: lastEventName } = [...eventNamesAndTypes].pop() ?? {}; // We need to cast eventNames to the last eventName defined in defineEmits
+        const typeCast =
+          eventNamesAndTypes.length > 1 && index + 1 < eventNamesAndTypes.length ? ` as '${lastEventName}'` : '';
+
+        return `
+    addEventListenerToElementRef(pdsComponentRef.value!, '${eventName}'${typeCast}, emit);`;
+      })
+      .join('');
+
+    const syncProperties = 'syncProperties(pdsComponentRef.value!, props);';
+
+    const additionalContent = `
+  ${props}
+  const pdsComponentRef = ref<${propsName} & HTMLElement>();${hasEvent ? defineEmits : ''}
```

Comment:
Inline interpolation could be avoided by joining and array of strings.  
Also why condition so late here? This could be taken care of earlier where `defineEmits` is initialized.